tests/integration/stp/02-stp-trunk.yml

```diff
@@ -0,0 +1,62 @@
+message: |
+  The devices under test form a layer-2 triangle of VLAN trunks
+
+  The bridge with the highest priority (s1) should become the root on all VLANs,
+  the link x1-x2 should get blocked. Please note that we're running traditional
+  STP where BPDUs are sent untagged.
+
+module: [ vlan, stp ]
+
+groups:
+  _auto_create: true
+  probes:
+    members: [ x1, x2 ]
+    device: eos
+
+nodes:
+  dut:
+    stp.priority: 4096  # High STP bridge priority (low value) -> becomes root
+  x2:
+    stp.priority: 16384
+
+vlans:
+  red:
+    mode: bridge
+    id: 42
+  blue:
+    mode: bridge
+    id: 76
+
+links:
+- dut:
+  x1:
+  vlan.trunk: [ red, blue ]
```

Comment:
Test relies on the native untagged VLAN 1 being allowed on the trunk, could make that explicit

Reply:
With the "legacy" STP, BPDUs are always sent untagged, regardless of what you configure on the trunk. See https://lostintransit.se/2024/07/16/encapsulation-of-pdus-on-trunk-ports/ for details